src/wallet/api/pending_transaction.cpp

```diff
@@ -188,6 +194,53 @@ std::vector<std::set<uint32_t>> PendingTransactionImpl::subaddrIndices() const
     return result;
 }
 
+std::string PendingTransactionImpl::multisigSignData() {
+    try {
+        if (!m_wallet.multisig().isMultisig) {
+            throw std::runtime_error("wallet is not multisig");
+        }
+
+        auto cipher = m_wallet.m_wallet->save_multisig_tx(m_pending_tx);
+        return epee::string_tools::buff_to_hex_nodelimer(cipher);
+    } catch (const std::exception& e) {
+        m_status = Status_Error;
```

Comment:
Should this not lock the new mutex (and others below) ?

Reply:
It seems `PendingTransactionImpl` class is not thread safe and there's no mutex at all. I just followed the same logic. If you want i can fix it